File: ENCO2018/SLAEL.cpp
#include<iostream>
#include<map>
#include<algorithm>
#include<iterator>
using namespace std;
int main(void){
	int t;cin>>t;
	while(t--){
		int n;cin>>n;
		long k;cin>>k;
		long a[n];
		map<long, long >dp;
		for(int i = 0; i < n; i++)cin>>a[i];
		for(int i = 0; i < n; i++){
			long first = 0, second = 0;
			for(int j = i; j < n; j++){
				if (a[j] > first){
					second = first;
					first = a[j];
				}
				else if (a[j] > second && a[j] != first)second = a[j];
				if((j - i) + 1 > dp[second])
				dp[second] = j - i + 1;
			}
		}
		long ans = 0;
		for(map<long,long>::iterator it=dp.begin(); it!=dp.end(); ++it){
			if(k >= it->first) ans = max(ans, it->second);
		}
		cout<<ans<<"\n";
	}
}

File: ENCO2018/SLAEL.py
for _ in range(int(input())):
	n,k = [int(x) for x in input().split()]
	a = [int(x) for x in input().split()]
	dp = {}
	# for i in range()
	for i in range(n):
		first,second = 0,0
		for j in range(i,n):
			if (a[j] > first):
				second = first
				first = a[j]
			elif (a[j] > second and a[j] != first):
				second = a[j]
			if dp.get(second):dp[second] = max(dp[second], (j - i) + 1)
			else: dp[second] = (j - i) + 1
		# print(second)
	ans = 0
	print(dp)
	for i in dp:
		if(k >= i):ans = max(ans, dp[i])
	print(ans)